nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/AbstractElasticsearchHttpProcessor.java

```diff
@@ -282,6 +291,14 @@ protected JsonNode parseJsonResponse(InputStream in) throws IOException {
         return mapper.readTree(in);
     }
 
+    @OnScheduled
+    public void setup(ProcessContext context) {
+        esUrlsIteratorAtomicReference.set(Iterables.cycle(Arrays.asList(context.getProperty(ES_URL).evaluateAttributeExpressions().getValue().split(","))).iterator());
```

Comment:
Better to `trim()` the values before setting them in the `Iterables.cycle` rather than every time they're retrieved in `getESUrl` - could create a `stream` from the `List` and apply a `map` function to `String::trim` each element before `collect`ing back into a `List` (or other similar approach)